exporters/prometheus/exporter.go

```diff
@@ -148,40 +152,48 @@ func (c *collector) Collect(ch chan<- prometheus.Metric) {
 				}
 				c.scopeInfos[scopeMetrics.Scope] = scopeInfo
 			}
+			c.mu.Unlock()
```

Comment:
This would not be called if `otel.Handle(err)` panics. I think we should do something like:

```go
func () {
  c.mu.Lock()
  defer c.mu.Unlock()
  // critical section
}()
```

Reply:
Oh, I see. You mean that if prometheus handler handles handles panic and then we have a deadlock here. I updated the code. PTAL.

Reply:
Yup.

The alternative way (probably more bulletproof) would be to move `otel.Handle()` call outside of the critical section. There is a pattern when for concurrent code to "avoid calling unknown/external code while inside a critical section".

This would be to proper fix as it would make having deadlock not possible.

Reply:
This is also done.